newlib/libc/tinystdio/bufio.c

```diff
@@ -307,7 +307,7 @@ __bufio_close(FILE *f)
          * FILE structs defined for stdin/stdout/stderr.
          */
         if (bf->bflags & __BFALL) {
-                bufio_close(bf);
+                ret = bufio_close(bf);
```

Comment:
This loses any potential error status return from the call to `__bufio_flush_locked`. Something like this should work?
```
if (bufio_close(bf) != 0)
    ret = EOF;
```

Reply:
`__bufio_flush_locked` always return 0.